app/assets/javascripts/govuk_publishing_components/analytics-ga4/ga4-core.js

```diff
@@ -211,6 +211,106 @@ window.GOVUK.analyticsGa4 = window.GOVUK.analyticsGa4 || {};
           }
         }
       }
+    },
+
+    ecommerceHelperFunctions: {
+      clearEcommerceObject: function() {
+        window.GOVUK.analyticsGa4.core.sendData({ search_results: { ecommerce: null } })
+      },
+
+      getIndex: function(data) {
+        var element = data.element
+        var startPosition = data.startPosition
+
+        return parseInt(element.getAttribute('data-ecommerce-index')) + startPosition - 1
+      },
+
+      getResultCount: function(element, id) {
+        // This returns a string e.g. '12,345 results'. Therefore to extract the number, we need to remove the comma and split the string at the space
+        var resultCount = element.querySelector('#' + id).textContent
+
+        if (!resultCount) {
+          return null
+        }
+
+        resultCount = resultCount.replace(',', '')
+        resultCount = resultCount.split(' ')[0]
+
+        return parseInt(resultCount)
+      },
+
+      populateEcommerceSchema: function(data) {
```

Comment:
This is quite a long and complex function. Would it be possible to break any parts of it out into separate functions to make it easier to read?

Reply:
I've been looking at simplifying this over the past few days and I think it's going to be a little tricky to simplify this so I'm thinking of adding this to the technical debt document and looking at this at a later date if you agree? Because I don't want it to be a blocker for this work being merged

Reply:
Yes, that's acceptable.